linkup-cli/src/env_files.rs

```diff
@@ -0,0 +1,280 @@
+    if let Ok(env_content) = fs::read_to_string(env_path) {
+        if env_content.contains(LINKUP_ENV_SEPARATOR) {
+            return Ok(());
+        }
+    }
+
+    let mut dev_env_content = fs::read_to_string(dev_env_path).map_err(|e| {
+        CliError::SetServiceEnv(
+            service.to_string(),
+            format!("could not read dev env file: {}", e),
+        )
+    })?;
+
+    if dev_env_content.ends_with('\n') {
+        dev_env_content.pop();
+    }
+
+    let mut env_file = OpenOptions::new()
+        .create(true)
+        .append(true)
+        .open(env_path)
+        .map_err(|e| {
+            CliError::SetServiceEnv(
+                service.to_string(),
+                format!("Failed to open .env file: {}", e),
+            )
+        })?;
+
+    let content = vec![
+        format!("\n{}", LINKUP_ENV_SEPARATOR),
+        format!("\n{}", dev_env_content),
+        format!("\n{}", LINKUP_ENV_SEPARATOR),
+    ];
+
+    writeln!(env_file, "{}", content.concat()).map_err(|e| {
+        CliError::SetServiceEnv(
+            service.to_string(),
+            format!("could not write to env file: {}", e),
+        )
+    })?;
+
+    Ok(())
+}
+
+pub fn clear_env_file(service: &str, env_path: &PathBuf) -> Result<()> {
+    let mut file_content = fs::read_to_string(env_path).map_err(|e| {
+        CliError::RemoveServiceEnv(
+            service.to_string(),
+            format!("could not read dev env file: {}", e),
+        )
+    })?;
+
+    let start_idx = file_content.find(LINKUP_ENV_SEPARATOR);
+    let end_idx = file_content.rfind(LINKUP_ENV_SEPARATOR);
+
+    if let (Some(mut start), Some(mut end)) = (start_idx, end_idx) {
+        if start < end {
+            let new_line_above_start =
+                start > 0 && file_content.chars().nth(start - 1) == Some('\n');
+            let new_line_bellow_end = file_content.chars().nth(end + 1) == Some('\n');
+
+            if new_line_above_start {
+                start -= 1;
+            }
+
+            if new_line_bellow_end {
+                end += 1;
+            }
+
+            file_content.drain(start..=end + LINKUP_ENV_SEPARATOR.len());
+        }
+
+        if file_content.ends_with('\n') {
+            file_content.pop();
+        }
+
+        // Write the updated content back to the file
+        let mut file = OpenOptions::new()
+            .write(true)
+            .truncate(true)
+            .open(env_path)
+            .map_err(|e| {
+                CliError::RemoveServiceEnv(
+                    service.to_string(),
+                    format!("Failed to open .env file for writing: {}", e),
+                )
+            })?;
+        file.write_all(file_content.as_bytes()).map_err(|e| {
+            CliError::RemoveServiceEnv(
+                service.to_string(),
+                format!("Failed to write .env file: {}", e),
+            )
+        })?;
+    }
+
+    Ok(())
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
Yeeesss this is really cool. Easier to write tests the way you managed to pull it out. Nice one!!